python/kserve/kserve/model_server.py

```diff
@@ -195,21 +194,21 @@ async def serve():
                                                   access_log_format=self.access_log_format)
                 await self._rest_server.run()
             else:
-                # Since py38 MacOS/Windows defaults to use spawn for starting multiprocessing.
-                # https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods
-                # Spawn does not work with FastAPI/uvicorn in multiprocessing mode, use fork for multiprocessing
-                # https://github.com/tiangolo/fastapi/issues/1586
                 serversocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
                 serversocket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
                 serversocket.bind(('0.0.0.0', self.http_port))
                 serversocket.listen(5)
-                multiprocessing.set_start_method('fork')
                 self._rest_server = UvicornServer(self.http_port, [serversocket],
                                                   self.dataplane, self.model_repository_extension,
                                                   self.enable_docs_url, log_config=self.log_config,
                                                   access_log_format=self.access_log_format)
+                # Since py38 MacOS/Windows defaults to use spawn for starting multiprocessing.
+                # https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods
+                # Spawn does not work with FastAPI/uvicorn in multiprocessing mode, use fork for multiprocessing
```

Comment:
Interestingly, uvicorn internally uses ['spawn' for creating new subprocesses](https://github.com/encode/uvicorn/blob/4f74ed144768d53fe6a959683c8e2a9bc51cc00a/uvicorn/_subprocess.py#L17).